src/services/stringCompletions.ts

```diff
@@ -513,12 +513,29 @@ namespace ts.Completions.StringCompletions {
                 }
             }
             if (!foundGlobal) {
-                forEachAncestorDirectory(scriptPath, ancestor => {
-                    const nodeModules = combinePaths(ancestor, "node_modules");
-                    if (tryDirectoryExists(host, nodeModules)) {
-                        getCompletionEntriesForDirectoryFragment(fragment, nodeModules, extensionOptions, host, /*exclude*/ undefined, result);
-                    }
-                });
+                const pnpapi = require("module").findPnpApi?.(scriptPath);
+
+                if (pnpapi) {
+                    try {
+                        // Splits a require request into its components, or return null if the request is a file path
+                        const pathRegExp = /^(?![a-zA-Z]:[\\/]|\\\\|\.{0,2}(?:\/|$))((?:@[^/]+\/)?[^/]+)\/*(.*|)$/;

```

Comment:
Would be worth looking if there's already a TS helper doing this (seems likely, perhaps in moduleNameResolver)

Reply:
Also, since the function is `getCompletionEntriesForNonRelativeModules`, are file paths actually possible?

Reply:
> are file paths actually possible?

It's not for filepaths, it's for getting the subpath out of `foo/bar` and `@foo/bar/baz`